x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/aggregation/blockhash/BlockHashTestCase.java

```diff
@@ -25,10 +47,163 @@ public abstract class BlockHashTestCase extends ESTestCase {
+                while (lookup.hasNext()) {
+                    try (IntBlock ords = lookup.next()) {
+                        for (int p = 0; p < ords.getPositionCount(); p++) {
+                            assertFalse(ords.isNull(p));
+                        }
+                    }
+                }
+            } finally {
+                Releasables.closeExpectNoException(keys);
+            }
+        }
+    }
+
+    protected void assertOrds(IntBlock ordsBlock, Integer... expectedOrds) {
+        assertOrds(ordsBlock, Arrays.stream(expectedOrds).map(l -> l == null ? null : new int[] { l }).toArray(int[][]::new));
+    }
+
+    protected void assertOrds(IntBlock ordsBlock, int[]... expectedOrds) {
+        assertEquals(expectedOrds.length, ordsBlock.getPositionCount());
+        for (int p = 0; p < expectedOrds.length; p++) {
+            int start = ordsBlock.getFirstValueIndex(p);
+            int count = ordsBlock.getValueCount(p);
+            if (expectedOrds[p] == null) {
+                if (false == ordsBlock.isNull(p)) {
+                    StringBuilder error = new StringBuilder();
+                    error.append(p);
+                    error.append(": expected null but was [");
+                    for (int i = 0; i < count; i++) {
+                        if (i != 0) {
+                            error.append(", ");
+                        }
+                        error.append(ordsBlock.getInt(start + i));
+                    }
+                    fail(error.append("]").toString());
+                }
+                continue;
+            }
+            assertFalse(p + ": expected not null", ordsBlock.isNull(p));
+            int[] actual = new int[count];
+            for (int i = 0; i < count; i++) {
+                actual[i] = ordsBlock.getInt(start + i);
+            }
+            assertThat("position " + p, actual, equalTo(expectedOrds[p]));
+        }
+    }
+
+    protected void assertKeys(Block[] actualKeys, Object... expectedKeys) {
+        Object[][] flipped = new Object[expectedKeys.length][];
+        for (int r = 0; r < flipped.length; r++) {
+            flipped[r] = new Object[] { expectedKeys[r] };
+        }
+        assertKeys(actualKeys, flipped);
+    }
+
+    protected void assertKeys(Block[] actualKeys, Object[][] expectedKeys) {
+        for (int r = 0; r < expectedKeys.length; r++) {
+            assertThat(actualKeys, arrayWithSize(expectedKeys[r].length));
+        }
+        for (int c = 0; c < actualKeys.length; c++) {
+            assertThat("block " + c, actualKeys[c].getPositionCount(), equalTo(expectedKeys.length));
+        }
+        for (int r = 0; r < expectedKeys.length; r++) {
+            for (int c = 0; c < actualKeys.length; c++) {
+                if (expectedKeys[r][c] == null) {
+                    assertThat("expected null key", actualKeys[c].isNull(r), equalTo(true));
+                    continue;
```

Comment:
This was a `return` before, which was skipping the checks on all further values. I fixed it, and a `BlockHashTest` case that failed because of it